internal/plugin/datasource_artifactory_item_test.go

```diff
@@ -163,7 +181,9 @@ items.find(
 			state.QueryTemplate.Set(qbuf.String())
 
 			cfg := template.Must(template.New("enos_data_artifactory_item").Parse(`data "enos_artifactory_item" "vault" {
+  {{if .Username.Value -}}
   username = "{{ .Username.Value }}"
```

Comment:
Sorta surprised that this works. I would have expected us to require setting username to null when the value doesn't exist.

Reply:
Ah, the datasource has the username marked as optional, as it should: https://github.com/hashicorp-forge/terraform-provider-enos/blob/main/internal/plugin/datasource_artifactory_item.go#L208